react/javascript/demo-external-api/README.md

```diff
@@ -0,0 +1,113 @@
+# Demo External API
+
+This example demonstrates usage of external API in an extension using javascript. Its uses OAUTH2 to get an access token from Google. It then use that token to create and update a spreadsheet using the Google sheet APIs. The key thing to observe are:
+
+- the OauthProvider which handles interfactions with Googles OAUTH2 identity service. The auth token is stored in the pushRouter state so that it survives a page reload.
+- The SheetsProvider which provides a simple abstraction to calling the Google sheets restful endpoints.
+
+## Prerequisites
+
+This demo requires a project with access to the sheets API and an OAUTH2 client to be setup in the Google console.
+
+1. [Open the Google console](https://console.cloud.google.com/).
+2. Select `IAM and Admin/Create a project`.
+3. Enter the name of the project and click `create` which navigates you to the project dashboard.
+4. On the project dashboard click `select project` for your newly created project. This displays information about your project and tasks you can do.
+5. In the `Getting started` section click `Explore and enable APIs`.
+6. Click `Libraries` and search for `Sheets`. This brings up a link to the `Google Sheets API`. `Select` it.
+7. Click `Enable` which brings you to the Google Sheets API setup.
+8. Click the `Configure Consent Screen` button.
```

Comment:
This is what I see after `Enable`
![image](https://user-images.githubusercontent.com/6099714/119886466-dc704480-bee7-11eb-9176-945a9aebe285.png)

So I think I need to create the credentials first?

Reply:
step 1
![image](https://user-images.githubusercontent.com/6099714/119886802-3a9d2780-bee8-11eb-9eae-c7019689e289.png)

Reply:
After `Done` I get 
![image](https://user-images.githubusercontent.com/6099714/119886986-6b7d5c80-bee8-11eb-8b00-3b9b5a3b93c7.png)

Reply:
I created some entries there, then I was able to proceed (for the most part the same way) with step 8.

Reply:
You should not have to do anything with Service accounts